spec/acceptance/upmark_spec.rb

```diff
@@ -173,6 +173,25 @@ def actual
 * Bullet 2
       MD
     end
+
+    specify 'converts nested ul as list' do
+      expect(<<-HTML.strip
+<ul>
+  <li>messenger</li>
+  <li>bag</li>
+  <ul>
```

Comment:
I don't think that's valid HTML. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul

The sub-`ul`s need to be themselves inside a `li`, so it'd be:

```
<ul>
  <li>messenger</li>
  <li>bag</li>
  <li> <!-- Opens here -->
    <ul>
      <li>electric</li>
      <li>skateboard</li>
    </ul>
  </li> <!-- Closes here -->
</ul>
```

Reply:
💯

Reply:
The input that jobs is receiving is in fact malformed HTML. But we should definitely support valid HTML. The code as it stands doesn't work for valid markup, so I'll look at that tomorrow.